---
title: "GWAX"
output: rmarkdown::html_vignette
description:
  skriv kort bekskrivelse her - ses ved siden artikler 
vignette: >
  %\VignetteIndexEntry{GWAX}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## GWAX

GWAX is short for genome-wide association study by proxy, which is an approach used to identify genomic variants that are statistically associated with a risk for a disease, exactly as GWAS (For more information read “”). The key difference is the case-control status, which GWAS uses a simple linear regression on. GWAX makes a proxy case-control status based on the individual´s family history. So if the individual case-control status is zero, meaning not having the disease, but just one the individual´s parents case-control status is one, meaning having the disease, the individual gets a proxy status as being one. GWAX therefore accounts for some family history and herabillity, which should give more useful information to the analysis. 

More information about the method of GWAX can be read in the article under references <https://pubmed.ncbi.nlm.nih.gov/28092683/>;



