---
title: "Getting Started with RyouSick"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Getting Started with RyouSick}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Simulation
To simulate genotype and case-control phenotype data using RyouSick you first have to simulate the disease parameters. Lets try to simulate disease parameters for genetic disease called bad_knee_syndrome on 10 SNPs with a heritability h2 of 0.8 and a prevalence in the population of 0.5:

```{r}
library("RyouSick")
bad_knee_syndrome <- sim_disease(n_snp = 100, h2 = 0.8, prevalence = 0.25)
bad_knee_syndrome

```

We can now chose what kind of simulation we want to perform. Either with family or without using the two functions:
 * `sim_genotypes_no_family()` simulates genotype and case-control phenotype without family case-control phenotypes.
 * `sim_genotypes_with_family()` simulates genotype and case-control phenotype with family case-control phenotypes.

Lets try to do a simulation of 10 genotypes under the badleg_syndrom disease parameters, with 1 to 3 siblings each: 

```{r}
bad_knee_data <- sim_genotypes_with_family(n = 100, disease = bad_knee_syndrome, path = "bad_knee_data2",n_sibs = c(1,2,3))

```

the information from the simulation can be accessed using dollar notation as such

```{r}
bad_knee_data$genotypes[]
bad_knee_data$FAM$Status
```

where we get the vector of case-control status for each individual. 

## Analysis
We can now perform analysis on our simulated data. The RyouSick package contains three methods for analysing the data: 
 * `GWAS()` Linear regression using genotype data and case-control status  
 * `GWAX()` Linear regression using genotype data and proxy case-control status  
 * `GWAX()` Linear regression using genotype data and posterior mean genetic liabilities
 
Lets now try each of the methods on our data:

```{r}
x <- GWAS(person = bad_knee_data, y = bad_knee_data$FAM$Status)
y <- GWAX(child = bad_knee_data)
z <- LTFH(child = bad_knee_data, h2 = 0.8, prevalence = 0.5)

```

Output will consist of estimated effect sizes, standard error, t-scores and pvalues for each SNP 

```{r}
x
```
Note that to access the analysis data for GWAX and LTFH you have to use `method_result$GWAS_Data`.

## Visualizations

We can then visualize the data using the function `Manhattan_plot()` to see which SNPs are determined to be causal: 

```{r}
Manhattan_plot(x)
```

If we want to visually compare the accumulated power for each of the methods, we can use the function `Power_plots()`. First we need to save the GWAS_data for each method in a list format.

```{r}
power_plot_data = list(x, y$GWAS_Data, z$GWAS_Data)
```

```{r}
Power_plots(power_plot_data, 5e-3)

```


rds.obj


## Prediction

As seen in `vignette("LTFH")Â´ the best method for evalutating causal SNPs is LTFH. So 



## Correct formatting of rds object

## Paralization

Since different users work on varo

