---
title: "Getting Started with RyouSick"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Getting Started with RyouSick}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Simulation
To illustrate the use of the package RyouSick lets go through an example, where we simulate a disease called bad-knee syndrome in a population.

Before simulating genotypes and case-control phenotypes data using RyouSick, we need to simulate the disease parameters with the function `sim_disease()`. The disease parameters for our target disorder is simulated on 10 SNPs with a heritability h2 of 0.8 and a prevalence in the population of 0.25:

```{r, eval = FALSE}
library("RyouSick")
bad_knee_syndrome <- sim_disease(n_snp = 10, h2 = 0.8, prevalence = 0.25)
```

```{r, include = FALSE}
#THIS BLOCK IS NOT SEEN IN KNITTED VERSION - external data loaded as seed is unusable in examples 
library("RyouSick")
bad_knee_data <- RyouSick::OpenRds(tools::file_path_sans_ext(system.file("extdata", "bad_knee_example_data.rds", package = "RyouSick")))
bad_knee_syndrome <- list(MAF = bad_knee_data$MAP$MAF, BETA = bad_knee_data$MAP$BETA, CAUSAL = ((bad_knee_data$MAP$BETA != 0) + 0), H2 = 0.8, PREVALENCE = 0.25, N_SNP = 10)

```

Disease parameters:
```{r}
bad_knee_syndrome
```

In the above output each MAF value at a given index indicates the frequency that the genome will vary from the reference genome. The CAUSAL vector and BETA vector likewise tell us which SNPs are simulated to be causally connected to developing the simulated disease and how big of an effect they have. In the above output we can see that SNP 4 has been simulated to be causal. 

We can now chose what kind of simulation we want to perform for our population. Either with family history or without using one of the two functions:

* `sim_genotypes_no_family()` simulates genotype and case-control phenotype without family case-control phenotypes
* `sim_genotypes_with_family()` simulates genotype and case-control phenotype with family case-control phenotypes

Lets try to use the simulation `sim_genotypes_with_family()`, with the bad_knee_syndrome disease parameters, and with 1 to 3 siblings for each individual and save it under the filename "bad_knee_data10x10": 

```{r, eval = FALSE}
bad_knee_data <- sim_genotypes_with_family(n = 10, 
                                           disease = bad_knee_syndrome, 
                                           path = "bad_knee_data10x10", 
                                           n_sibs = c(1,2,3))
```

The bad_knee_data will now contain a list object, which we will refer to as an rds object, with a reference to a file-backed matrix (matrix stored on disk), a tibble containing information on the disease (MAP) and a tibble containing phenotype information on the genotypes. Individual columns in the data can be accessed using the dollar notation.

```{r}
bad_knee_data$FAM$Status
```
Slices of the genotypes can be accessed as well.
```{r}
bad_knee_data$genotypes[1:10, 1:10]

```
Here each row represents an individuals genotypes and each column a SNP. 
(WARNING: we highly advise against indexing all the data at once using [ ], as this will try to load the whole file-backed matrix into memory, which, depending on the size of the file, might cause problems)

All of the simulated information contained in our example case bad_knee_data can be seen below.

```{r}
knitr::kable(bad_knee_data$MAP, align = "ccc")
knitr::kable(bad_knee_data$FAM, align = "ccccccc")
```


## Analysis
We now perform analysis on our simulated data. The RyouSick package contains three methods for analyzing the data: 

 * `GWAS()` Linear regression using genotype data and case-control status  
 * `GWAX()` Linear regression using genotype data and proxy case-control status  
 * `LTFH()` Linear regression using genotype data and posterior mean genetic liabilities
 
Lets now try each of the methods on our data:

```{r, warning=FALSE}
x <- GWAS(rds.obj = bad_knee_data, y = bad_knee_data$FAM$Status)
y <- GWAX(rds.obj = bad_knee_data)
z <- LTFH(rds.obj = bad_knee_data, h2 = 0.8, prevalence = 0.5)

```

The output will consist of estimated effect sizes, standard errors, t-scores and p-values for each SNP, which again can be accessed using dollar notation.

```{r}
knitr::kable(x)
```
Note that to access the analysis data for GWAX and LTFH you have to use `method_result$GWAS_Data$`.

Statistics for the data can be calculated by passing the p-values, the true causal SNPs and a significance level to the `get_stats()` function

```{r}
gwas_stats <- get_stats(pvalues = x$p.value, 
                        true_causal = bad_knee_syndrome$CAUSAL, 
                        a = 0.05)
knitr::kable(gwas_stats)
```


## Visualizations

We can then visualize the data using the function `Manhattan_plot()` to get a picture of which SNPs and where in the genome they have been determined to be causal: 

```{r}
Manhattan_plot(x, a = 5e-7)
```


If we want to visually compare the accumulated power for each of the methods, we can use the function `Power_plots()`. We basically want to see which method finds most causal SNP´s, and that the SNP´s are actually predetermined to be causal. To check for this we use the output from `get_stats()` which tells us about the frequency for false positives for each method. 

Before we use the function `Power_plots()` we need to save the GWAS_data for each method in a list format. For illustrative purposes we set the significance level to 0.5. 

```{r}
power_plot_data = list(x, y$GWAS_Data, z$GWAS_Data)
```

With the data saved, we can now use `Power_plots()`:

```{r}
Power_plots(Gwas_data = power_plot_data, a = 0.5)
```

For output from `Power_plots()` we note that the line for GWAS cannot be seen in the above plot because it lays under the plot for GWAX, which, when using such a low sample size, produces almost identical results to GWAS.

## Prediction

As seen in `vignette("LTFH")` the best method for evaluating causal SNPs is LT-FH. So let's try to use `Prediction_cross_validation()` on our LT-FH data to get a model that can predict future cases.

```{r, warning=FALSE, include = FALSE, message = FALSE}
model_results <- Prediction_cross_validation(rds.obj = bad_knee_data,
                                             k = 2,
                                             method = "LTFH")
```

```{r, eval=FALSE}
model_results <- Prediction_cross_validation(rds.obj = bad_knee_data,
                                             k = 2,
                                             method = "LTFH")
```

This model can now be used in the `Predict_status()` function to predict status for future genotypes. For illustrative purposes we will use our old data again, as the input needs to be formatted as an rds object.

```{r, warning=FALSE, include = FALSE, message = FALSE}
predicted_status <- Predict_status(rds.obj = bad_knee_data,
                                   model = model_results$Best_Model,
                                   prevalence = 0.25)
```


```{r, eval=FALSE}
predicted_status <- Predict_status(rds.obj = bad_knee_data,
                                   model = model_results$Best_Model,
                                   prevalence = 0.25)
```
This gives us the following results.

```{r}
predicted_status
```


## Correct formatting of rds object

If you wish to run the analysis methods on your own data it must first be formatted properly. To do so you must first initialize an empty rds file and object matching the size of your data using the `createRds()` function.

```{r, eval=FALSE}
user_input_obj <- createRds("user_input_data", nrow = 10, ncol = 10)
```

The rds object that is returned can then be filled up by adding columns/vectors in the MAP and FAM tibbles using dollar notation. To integrate with the methods in the package, columns must be given specific names based on their contents. The required information to run must functions and their naming is listed below

* MAF (in MAP):  vector with minor allele frequencies for SNPs
* Status (in FAM):  vector with status for genotypes
* p1_Status (in FAM):  vector with parent 1 status for genotypes (only used in GWAX and LTFH)
* p2_Status (in FAM):  vector with parent 2 status for genotypes (only used in GWAX and LTFH)
* sibs_Status (in FAM):  list containing a vector with sibling status for each genotype or NULL if no siblings for genotype (only used in LTFH)

Note that the names are case-sensitive. An example of filling up the genotypes with a matrix consisting of only 1's and creating a Status column can be seen below. Remember that when making changes to the tibbles FAM and MAP, the changes to the rds object must be saved using `saveRds()`.

```{r, eval =FALSE}
user_input_obj$genotypes[1:10,1:10] <- matrix(1, ncol = 10, nrow = 10)
user_input_obj$Status <- c(1,1,1,0,0,0,0,1,0,0)
saveRds(user_input_obj)
```

Additionally, it is recommended that the file-backed matrix be filled in blocks if working with large quantities of data (read more in article `vignette("Writing_efficient_R_code_and_working_with_large_datasets")`). The rds file can later be opened again using the `OpenRds()` function.


## Parallelization

Since most users work on very different computers with various setups Parallelization is not enabled by default. To run simulations using Parallelization, which is recommended as it provides a significant speedup in most cases, you must first set a Parallelization plan. For most basic computers running on windows the Parallelization plan below will work just fine. 

```{r, eval = FALSE}
future::plan(future::multisession, 
             workers = max(1, future::availableCores(logical = F) - 1))
```

The function should be called once at the beginning before any simulations have been run.

We refer the reader to <https://future.futureverse.org/reference/plan.html> and the future backend to read up on on how to set up a plan that might work best on their setup. 



