<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Overview of the LT-FH method, results from a run on a simulated 100.000x100.000 dataset and comparison with runs using GWAS and GWAX.">
<title>LTFH • RyouSick</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="LTFH">
<meta property="og:description" content="Overview of the LT-FH method, results from a run on a simulated 100.000x100.000 dataset and comparison with runs using GWAS and GWAX.">
<meta property="og:image" content="https://best-nicklas.github.io/Dataprojekt/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">RyouSick</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/RyouSick.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../articles/index.html">Articles</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="LTFH_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>LTFH</h1>
            
      
      
      <div class="d-none name"><code>LTFH.Rmd</code></div>
    </div>

    
    
<p>LT-FH is short for liability threshold model, conditional on case-control status and family history (original paper by Hujoel, M.L.A. et al (2020)). The approach in LT-FH is to assign posterior mean genetic liabilities for each individual given that prior information on case-control status for themselves and family (their configuration) is already known. This posterior liability can then be used to perform GWAS alongside genotyped data, giving us more information than what a two-leveled case-control status would. More broadly, LTFH expands upon the liability threshold model that was touched upon in the <code><a href="../articles/Simulation_Basis.html">vignette("Simulation_Basis")</a></code>. In LTFH we now model the liabilities of our target individuals using the following multivariate normal distribution that takes the liabilities of family members into account:</p>
<p><span class="math display">\[\begin{bmatrix}
l_g \\
l_o \\
l_{p1} \\
l_{p2} \\
l_s \\
\end{bmatrix} \sim MVN_5\left( \begin{bmatrix}
0 \\
0 \\
0 \\
0 \\
0 \\
\end{bmatrix},\begin{bmatrix}
h^2 &amp; h^2 &amp;0.5h^2 &amp; 0.5h^2&amp; 0.5h^2\\
h^2 &amp; 1 &amp; 0.5h^2&amp; 0.5h^2&amp; 0.5h^2\\
0 &amp;0.5h^2 &amp; 1 &amp; 0 &amp; 0.5h^2\\
0 &amp; 0.5h^2 &amp; 0 &amp; 1 &amp; 0.5h^2\\
0 &amp; 0.5h^2&amp;0.5h^2 &amp;0.5h^2 &amp; 1\\
\end{bmatrix}\right)\]</span></p>
<p>Here <span class="math inline">\(l_g\)</span> is the target individuals genetic liability, <span class="math inline">\(l_o\)</span> their full liablilty, <span class="math inline">\(l_{p1}\)</span> the full liability of parent 1, <span class="math inline">\(l_{p2}\)</span> the full liability of parent 2 and <span class="math inline">\(l_s\)</span> the liability of a sibling. In the above the liabilities have been modeled using only 1 sibling. The model can be extended with more siblings by adding another row and column containing <span class="math inline">\(0.5h^2\)</span> in every entrance except the diagonal entrance which should be 1.</p>
<p>To calculate the posterior mean genetic liabilities RyouSick first Gibbs samples genetic liabilities from the above distribution. This specific approach is a light implementation of method for calculating posterior mean genetic liabilites outlined the LT-FH++ method (see Pedersen, E.M. et al (2022)). It takes advantage of the fact that by calculating conditional liabilities in the Gibbs sampling process, fx. <span class="math inline">\(l_g\)</span> conditional on <span class="math inline">\(l_o, l_{p1},l_{p2} \text{ and }l_s\)</span>, we can get samples and sample means from our multivariate distribution much easier than using integration. In more detail we take case-control status into account by sampling from truncated conditional normal distributions. This kind of distribution is derived from that of a normally distributed random variable by bounding it from either below or above. When applying the information about the case-control status, we are able to bound the distribution from below when the liability for an individual is a case, and from above when not a case. Once the Gibbs sampling process converges to the target distribution for our configuration, which we check by evaluating the SEM (standard error of the mean), we begin to sample the genetic liabilities from the converged distribution. These are then used to calculate our posterior mean genetic liability for the configuration.</p>
<p>Since we only have to sample a mean genetic liability for each configuration, which is made from the status of parents, child and siblings, and not for every genotype, as two genotypes can have the same configuration, posterior mean genetic liabilities can be calculated relatively quickly. When the posterior mean genetic liabilities have been calculated for all configurations available in our data we match it to the individuals in our dataset and finally perform GWAS on the genotypes and the posterior mean genetic liabilities.</p>
<p>Using the LT-FH approach on the same simulated data that we analyzed in the GWAS and GWAX articles we get the following results, with results from GWAS and GWAX included for comparison.</p>
<p>For GWAS</p>
<p><img src="../reference/figures/GWAS.PNG"><img src="../reference/figures/GWAS_stats.PNG" style="width:100.0%"> For GWAX <img src="../reference/figures/GWAX.PNG"><img src="../reference/figures/GWAX_stats.PNG" style="width:100.0%"> For LTFH</p>
<p><img src="../reference/figures/LTFH.PNG"><img src="../reference/figures/LTFH_stats.PNG" style="width:100.0%"></p>
<p>Addtionally we get the following power plots:</p>
<p><img src="../reference/figures/power_plot.PNG"></p>
<p>We see that the LT-FH method as expected performs better at finding true positives than both GWAS and GWAX.</p>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li><p>Hujoel, M.L.A., Gazal, S., Loh, P. et al. Liability threshold modeling of case-control status and family history of disease increases association power. Nat Genet, 52(5) p541-547 (2020). <a href="https://pubmed.ncbi.nlm.nih.gov/32313248/" class="external-link uri">https://pubmed.ncbi.nlm.nih.gov/32313248/</a></p></li>
<li><p>Pedersen, E.M., Agerbo, E., Plana-Ripoll, et al. Accounting for age of onset and family history improves power in genome-wide association studies. The American Journal of Human Genetics, Volume 109, issue 3, P417-432 (2022). <a href="https://www.cell.com/ajhg/fulltext/S0002-9297(22)00009-X/" class="external-link uri">https://www.cell.com/ajhg/fulltext/S0002-9297(22)00009-X/</a></p></li>
</ul>
</div>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicklas Stanislaus, Markus Nielsen, Christian Mains.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
