<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="RyouSick">
<title>Getting Started with RyouSick • RyouSick</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting Started with RyouSick">
<meta property="og:description" content="RyouSick">
<meta property="og:image" content="https://best-nicklas.github.io/Dataprojekt/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">RyouSick</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/RyouSick.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/index.html">Articles</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="RyouSick_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Getting Started with RyouSick</h1>
            
      
      
      <div class="d-none name"><code>RyouSick.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="simulation">Simulation<a class="anchor" aria-label="anchor" href="#simulation"></a>
</h2>
<p>To illustrate the use of RyouSick lets go through an example where we try to simulate a disease called bad-knee syndrome in a population.</p>
<p>When trying to simulate genotype and case-control phenotype data using RyouSick you first have to simulate the disease parameters. Disease parameters for our target disorder is simulated on 10 SNPs with a heritability h2 of 0.8 and a prevalence in the population of 0.25:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://best-nicklas.github.io/Dataprojekt/">"RyouSick"</a></span><span class="op">)</span>
<span class="va">bad_knee_syndrome</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sim_disease.html">sim_disease</a></span><span class="op">(</span>n_snp <span class="op">=</span> <span class="fl">10</span>, h2 <span class="op">=</span> <span class="fl">0.8</span>, prevalence <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></code></pre></div>
<p>which outputs</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bad_knee_syndrome</span>
<span class="co">#&gt; $MAF</span>
<span class="co">#&gt;  [1] 0.40071281 0.22309419 0.04419435 0.42861384 0.46978992 0.15051858</span>
<span class="co">#&gt;  [7] 0.15620303 0.11896250 0.07729138 0.20345860</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $BETA</span>
<span class="co">#&gt;  [1]  0.000000  0.000000  0.000000 -0.296382  0.000000  0.000000  0.000000</span>
<span class="co">#&gt;  [8]  0.000000  0.000000  0.000000</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $CAUSAL</span>
<span class="co">#&gt;  [1] 0 0 0 1 0 0 0 0 0 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $H2</span>
<span class="co">#&gt; [1] 0.8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $PREVALENCE</span>
<span class="co">#&gt; [1] 0.25</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $N_SNP</span>
<span class="co">#&gt; [1] 10</span></code></pre></div>
<p>In the above output each MAF value at a given index indicates the frequency at which the genome will vary from the reference. The CAUSAL vector and BETA vector likewise tell us which SNPs are simulated to be causally connected to developing the simulated disease and how big of an effect they have. In the above output we are shown that SNP 4 has been simulated to be causal.</p>
<p>We can now chose what kind of simulation we want to perform for our population. Either with family history or without using one of the two functions:</p>
<ul>
<li>
<code><a href="../reference/sim_genotypes_no_family.html">sim_genotypes_no_family()</a></code> simulates genotype and case-control phenotype without family case-control phenotypes</li>
<li>
<code><a href="../reference/sim_genotypes_with_family.html">sim_genotypes_with_family()</a></code> simulates genotype and case-control phenotype with family case-control phenotypes</li>
</ul>
<p>Lets try to do a simulation of 10 genotypes under the bad_knee_syndrome disease parameters with 1 to 3 siblings for each genotype and save it under the filename “bad_knee_data10x10”:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bad_knee_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sim_genotypes_with_family.html">sim_genotypes_with_family</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span>, 
                                           disease <span class="op">=</span> <span class="va">bad_knee_syndrome</span>, 
                                           path <span class="op">=</span> <span class="st">"bad_knee_data10x10"</span>, 
                                           n_sibs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>The bad_knee_data will now contain a list object, which we will refer to as an rds object, with a reference to a file-backed matrix (matrix stored on disk), a tibble containing information on the disease (MAP) and a tibble containing phenotype information on the genotypes. Individual columns in the data can be accessed using the dollar notation.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bad_knee_data</span><span class="op">$</span><span class="va">FAM</span><span class="op">$</span><span class="va">Status</span>
<span class="co">#&gt;       [,1]</span>
<span class="co">#&gt;  [1,]    1</span>
<span class="co">#&gt;  [2,]    0</span>
<span class="co">#&gt;  [3,]    0</span>
<span class="co">#&gt;  [4,]    0</span>
<span class="co">#&gt;  [5,]    0</span>
<span class="co">#&gt;  [6,]    1</span>
<span class="co">#&gt;  [7,]    0</span>
<span class="co">#&gt;  [8,]    0</span>
<span class="co">#&gt;  [9,]    1</span>
<span class="co">#&gt; [10,]    0</span></code></pre></div>
<p>Slices of the genotype data can be accessed as well.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bad_knee_data</span><span class="op">$</span><span class="va">genotypes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>
<span class="co">#&gt;       [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10]</span>
<span class="co">#&gt;  [1,]    0    2    0    0    0    0    0    0    0     1</span>
<span class="co">#&gt;  [2,]    1    0    1    2    0    0    0    1    0     1</span>
<span class="co">#&gt;  [3,]    0    0    0    1    1    0    0    1    1     1</span>
<span class="co">#&gt;  [4,]    0    1    0    0    1    0    0    1    0     0</span>
<span class="co">#&gt;  [5,]    0    1    0    0    0    0    0    1    0     0</span>
<span class="co">#&gt;  [6,]    0    0    0    0    0    1    1    0    0     1</span>
<span class="co">#&gt;  [7,]    1    1    0    2    1    1    1    0    0     1</span>
<span class="co">#&gt;  [8,]    0    0    0    2    1    0    0    0    0     0</span>
<span class="co">#&gt;  [9,]    0    0    0    0    0    0    0    1    0     1</span>
<span class="co">#&gt; [10,]    0    0    0    2    1    0    0    0    0     0</span></code></pre></div>
<p>Here each row represents a genotype and each column a SNP. (WARNING: we highly advise against indexing all the data at once using [ ], as this will try to load the whole file-backed matrix into memory, which, depending on the size of the file, might cause problems)</p>
<p>All of the simulated information contained in our example case bad_knee_data can be seen below.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">bad_knee_data</span><span class="op">$</span><span class="va">MAP</span>, align <span class="op">=</span> <span class="st">"ccc"</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="center">SNP_ID</th>
<th align="center">MAF</th>
<th align="center">BETA</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">1</td>
<td align="center">0.4007128</td>
<td align="center">0.000000</td>
</tr>
<tr class="even">
<td align="center">2</td>
<td align="center">0.2230942</td>
<td align="center">0.000000</td>
</tr>
<tr class="odd">
<td align="center">3</td>
<td align="center">0.0441943</td>
<td align="center">0.000000</td>
</tr>
<tr class="even">
<td align="center">4</td>
<td align="center">0.4286138</td>
<td align="center">-0.296382</td>
</tr>
<tr class="odd">
<td align="center">5</td>
<td align="center">0.4697899</td>
<td align="center">0.000000</td>
</tr>
<tr class="even">
<td align="center">6</td>
<td align="center">0.1505186</td>
<td align="center">0.000000</td>
</tr>
<tr class="odd">
<td align="center">7</td>
<td align="center">0.1562030</td>
<td align="center">0.000000</td>
</tr>
<tr class="even">
<td align="center">8</td>
<td align="center">0.1189625</td>
<td align="center">0.000000</td>
</tr>
<tr class="odd">
<td align="center">9</td>
<td align="center">0.0772914</td>
<td align="center">0.000000</td>
</tr>
<tr class="even">
<td align="center">10</td>
<td align="center">0.2034586</td>
<td align="center">0.000000</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">bad_knee_data</span><span class="op">$</span><span class="va">FAM</span>, align <span class="op">=</span> <span class="st">"ccccccc"</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="center">ID</th>
<th align="center">Genetic_Liability</th>
<th align="center">Full_Liability</th>
<th align="center">Status</th>
<th align="center">p1_Status</th>
<th align="center">p2_Status</th>
<th align="center">sibs_Status</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="center">1</td>
<td align="center">0.36302377</td>
<td align="center">1.0514806</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">1</td>
</tr>
<tr class="even">
<td align="center">2</td>
<td align="center">-0.48394788</td>
<td align="center">-1.2211161</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0, 0, 0</td>
</tr>
<tr class="odd">
<td align="center">3</td>
<td align="center">-0.06046206</td>
<td align="center">0.2717148</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0, 0</td>
</tr>
<tr class="even">
<td align="center">4</td>
<td align="center">0.36302377</td>
<td align="center">0.6714890</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">1</td>
</tr>
<tr class="odd">
<td align="center">5</td>
<td align="center">0.36302377</td>
<td align="center">-0.2015389</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0, 1</td>
</tr>
<tr class="even">
<td align="center">6</td>
<td align="center">0.36302377</td>
<td align="center">0.7119287</td>
<td align="center">1</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">0, 1, 1</td>
</tr>
<tr class="odd">
<td align="center">7</td>
<td align="center">-0.48394788</td>
<td align="center">-0.6437396</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0, 0, 0</td>
</tr>
<tr class="even">
<td align="center">8</td>
<td align="center">-0.48394788</td>
<td align="center">-0.3891188</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
</tr>
<tr class="odd">
<td align="center">9</td>
<td align="center">0.36302377</td>
<td align="center">1.3064947</td>
<td align="center">1</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0, 0</td>
</tr>
<tr class="even">
<td align="center">10</td>
<td align="center">-0.48394788</td>
<td align="center">-0.2376695</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0, 1, 1</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="analysis">Analysis<a class="anchor" aria-label="anchor" href="#analysis"></a>
</h2>
<p>We now perform analysis on our simulated data. The RyouSick package contains three methods for analyzing the data:</p>
<ul>
<li>
<code><a href="../reference/GWAS.html">GWAS()</a></code> Linear regression using genotype data and case-control status<br>
</li>
<li>
<code><a href="../reference/GWAX.html">GWAX()</a></code> Linear regression using genotype data and proxy case-control status<br>
</li>
<li>
<code><a href="../reference/LTFH.html">LTFH()</a></code> Linear regression using genotype data and posterior mean genetic liabilities</li>
</ul>
<p>Lets now try each of the methods on our data:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWAS.html">GWAS</a></span><span class="op">(</span>rds.obj <span class="op">=</span> <span class="va">bad_knee_data</span>, y <span class="op">=</span> <span class="va">bad_knee_data</span><span class="op">$</span><span class="va">FAM</span><span class="op">$</span><span class="va">Status</span><span class="op">)</span>
<span class="va">y</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/GWAX.html">GWAX</a></span><span class="op">(</span>rds.obj <span class="op">=</span> <span class="va">bad_knee_data</span><span class="op">)</span>
<span class="va">z</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/LTFH.html">LTFH</a></span><span class="op">(</span>rds.obj <span class="op">=</span> <span class="va">bad_knee_data</span>, h2 <span class="op">=</span> <span class="fl">0.8</span>, prevalence <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></code></pre></div>
<p>Output will consist of estimated effect sizes, standard errors, t-scores and p-values for each SNP, which again can be accessed using dollar notation.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="right">estim</th>
<th align="right">std.err</th>
<th align="right">score</th>
<th align="right">p.value</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right">-0.3750000</td>
<td align="right">0.3827328</td>
<td align="right">-0.9797959</td>
<td align="right">0.3558838</td>
</tr>
<tr class="even">
<td align="right">0.1111111</td>
<td align="right">0.2383068</td>
<td align="right">0.4662524</td>
<td align="right">0.6534672</td>
</tr>
<tr class="odd">
<td align="right">-0.3333333</td>
<td align="right">0.5270463</td>
<td align="right">-0.6324555</td>
<td align="right">0.5447373</td>
</tr>
<tr class="even">
<td align="right">-0.3033708</td>
<td align="right">0.1341274</td>
<td align="right">-2.2618111</td>
<td align="right">0.0535707</td>
</tr>
<tr class="odd">
<td align="right">-0.6000000</td>
<td align="right">0.2449490</td>
<td align="right">-2.4494897</td>
<td align="right">0.0399685</td>
</tr>
<tr class="even">
<td align="right">0.2500000</td>
<td align="right">0.3952847</td>
<td align="right">0.6324555</td>
<td align="right">0.5447373</td>
</tr>
<tr class="odd">
<td align="right">0.2500000</td>
<td align="right">0.3952847</td>
<td align="right">0.6324555</td>
<td align="right">0.5447373</td>
</tr>
<tr class="even">
<td align="right">-0.2000000</td>
<td align="right">0.3162278</td>
<td align="right">-0.6324555</td>
<td align="right">0.5447373</td>
</tr>
<tr class="odd">
<td align="right">-0.3333333</td>
<td align="right">0.5270463</td>
<td align="right">-0.6324555</td>
<td align="right">0.5447373</td>
</tr>
<tr class="even">
<td align="right">0.5000000</td>
<td align="right">0.2795085</td>
<td align="right">1.7888544</td>
<td align="right">0.1114343</td>
</tr>
</tbody>
</table>
<p>Note that to access the analysis data for GWAX and LTFH you have to use <code>method_result$GWAS_Data$</code>.</p>
<p>Statistics for the data can be calculated by passing the p-values, the true causal SNPs and a significance level to the <code><a href="../reference/get_stats.html">get_stats()</a></code> function</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gwas_stats</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_stats.html">get_stats</a></span><span class="op">(</span>pvalues <span class="op">=</span> <span class="va">x</span><span class="op">$</span><span class="va">p.value</span>, 
                        true_causal <span class="op">=</span> <span class="va">bad_knee_syndrome</span><span class="op">$</span><span class="va">CAUSAL</span>, 
                        a <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span>
<span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">gwas_stats</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">false_negatives</th>
<th align="right">true_negatives</th>
<th align="right">false_positives</th>
<th align="right">true_positives</th>
<th align="right">total</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Count</td>
<td align="right">1.0</td>
<td align="right">8.0</td>
<td align="right">1.0</td>
<td align="right">0</td>
<td align="right">10</td>
</tr>
<tr class="even">
<td align="left">Frequency</td>
<td align="right">0.1</td>
<td align="right">0.8</td>
<td align="right">0.1</td>
<td align="right">0</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="visualizations">Visualizations<a class="anchor" aria-label="anchor" href="#visualizations"></a>
</h2>
<p>We can then visualize the data using the function <code><a href="../reference/Manhattan_plot.html">Manhattan_plot()</a></code> to get a picture of which SNPs and where in the genome they have been determined to be causal:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/Manhattan_plot.html">Manhattan_plot</a></span><span class="op">(</span><span class="va">x</span>, a <span class="op">=</span> <span class="fl">5e-7</span><span class="op">)</span></code></pre></div>
<p><img src="RyouSick_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<p>If we want to visually compare the accumulated power for each of the methods, we can use the function <code><a href="../reference/Power_plots.html">Power_plots()</a></code>. The power for each of the methods gives us an idea of which of the methods, that finds most of the causal SNP´s. Since we randomly assign causal SNP´s when we simulate disease parameters, it´s possible to calculate the amount of false positives each of the methods have, which ensures us, that the methods does not analyze SNP´s to be causal, that are predetermined not to be causal.</p>
<p>Before we use the the function <code><a href="../reference/Power_plots.html">Power_plots()</a></code> we need to save the GWAS_data for each method in a list format. For illustrative purposes we set the significance level to 0.5.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">power_plot_data</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span><span class="op">$</span><span class="va">GWAS_Data</span>, <span class="va">z</span><span class="op">$</span><span class="va">GWAS_Data</span><span class="op">)</span>
<span class="fu"><a href="../reference/Power_plots.html">Power_plots</a></span><span class="op">(</span>Gwas_data <span class="op">=</span> <span class="va">power_plot_data</span>, a <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></code></pre></div>
<p><img src="RyouSick_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
<p>We also note that the line for GWAS cannot be seen in the above plot because it lays under the plot for GWAX, which, when using such a low sample size, produces almost identical results to GWAS.</p>
</div>
<div class="section level2">
<h2 id="prediction">Prediction<a class="anchor" aria-label="anchor" href="#prediction"></a>
</h2>
<p>As seen in <code><a href="../articles/LTFH.html">vignette("LTFH")</a></code> the best method for evaluating causal SNPs is LT-FH. So let’s try to use <code><a href="../reference/Prediction_cross_validation.html">Prediction_cross_validation()</a></code> on our LT-FH data to get a model that can predict future cases.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">model_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Prediction_cross_validation.html">Prediction_cross_validation</a></span><span class="op">(</span>rds.obj <span class="op">=</span> <span class="va">bad_knee_data</span>,
                                             k <span class="op">=</span> <span class="fl">2</span>,
                                             method <span class="op">=</span> <span class="st">"LTFH"</span><span class="op">)</span></code></pre></div>
<p>This model can now be used in the <code><a href="../reference/Predict_status.html">Predict_status()</a></code> function to predict status for future genotypes. For illustrative purposes we will use our old data again, as the input needs to be formatted as an rds object.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">predicted_status</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/Predict_status.html">Predict_status</a></span><span class="op">(</span>rds.obj <span class="op">=</span> <span class="va">bad_knee_data</span>,
                                   model <span class="op">=</span> <span class="va">model_results</span><span class="op">$</span><span class="va">Best_Model</span>,
                                   prevalence <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></code></pre></div>
<p>This gives us the following results.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">predicted_status</span>
<span class="co">#&gt;  [1] 1 0 0 0 0 1 0 0 0 0</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="correct-formatting-of-rds-object">Correct formatting of rds object<a class="anchor" aria-label="anchor" href="#correct-formatting-of-rds-object"></a>
</h2>
<p>If you wish to run the analysis methods on your own data it must first be formatted properly. To do so you must first initialize an empty .rds file and object matching the size of your data using the <code><a href="../reference/createRds.html">createRds()</a></code> function.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">user_input_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/createRds.html">createRds</a></span><span class="op">(</span><span class="st">"user_input_data"</span>, nrow <span class="op">=</span> <span class="fl">10</span>, ncol <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></code></pre></div>
<p>The rds object that is returned can then be filled up by adding columns/vectors in the MAP and FAM tibbles using dollar notation. To integrate with the methods in the package, columns must be given specific names based on their contents. The required information to run must functions and their naming is listed below</p>
<ul>
<li>MAF (in MAP): vector with minor allele frequencies for SNPs</li>
<li>Status (in FAM): vector with status for genotypes</li>
<li>p1_Status (in FAM): vector with parent 1 status for genotypes (only used in GWAX and LTFH)</li>
<li>p2_Status (in FAM): vector with parent 2 status for genotypes (only used in GWAX and LTFH)</li>
<li>sibs_Status (in FAM): list containing a vector with sibling status for each genotype or NULL if no siblings for genotype (only used in LTFH)</li>
</ul>
<p>Note that the names are case-sensitive. An example of filling up the genotypes with a matrix consisting of only 1’s and creating a Status column can be seen below. Remember that when making changes to the tibbles FAM and MAP, the changes to the rds object must be saved using <code><a href="../reference/saveRds.html">saveRds()</a></code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">user_input_obj</span><span class="op">$</span><span class="va">genotypes</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">1</span>, ncol <span class="op">=</span> <span class="fl">10</span>, <span class="va">nrow</span><span class="op">)</span>
<span class="va">user_input_obj</span><span class="op">$</span><span class="va">Status</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">1</span>,<span class="fl">0</span>,<span class="fl">0</span>,<span class="fl">0</span>,<span class="fl">0</span>,<span class="fl">1</span>,<span class="fl">0</span>,<span class="fl">0</span><span class="op">)</span>
<span class="fu"><a href="../reference/saveRds.html">saveRds</a></span><span class="op">(</span><span class="va">user_input_obj</span><span class="op">)</span></code></pre></div>
<p>Additionally, it is recommended that the file-backed matrix be filled in blocks if working with large quantities of data (read more in article <code><a href="../articles/Writing_efficient_R_code_and_working_with_large_datasets.html">vignette("Writing_efficient_R_code_and_working_with_large_datasets")</a></code>). The rds file can later be opened again using the <code><a href="../reference/OpenRds.html">OpenRds()</a></code> function.</p>
</div>
<div class="section level2">
<h2 id="parallelization">Parallelization<a class="anchor" aria-label="anchor" href="#parallelization"></a>
</h2>
<p>Since most users work on very different computers with various setups Parallelization is not enabled by default. To run simulations using Parallelization, which is recommended as it provides a significant speedup in most cases, you must first set a Parallelization plan. For most basic computers running on windows the Parallelization plan below will work just fine.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">future</span><span class="fu">::</span><span class="fu"><a href="https://future.futureverse.org/reference/plan.html" class="external-link">plan</a></span><span class="op">(</span><span class="fu">future</span><span class="fu">::</span><span class="va"><a href="https://future.futureverse.org/reference/multisession.html" class="external-link">multisession</a></span>, 
             workers <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu">future</span><span class="fu">::</span><span class="fu"><a href="https://future.futureverse.org/reference/re-exports.html" class="external-link">availableCores</a></span><span class="op">(</span>logical <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>The function should be called once at the beginning before any simulations have been run.</p>
<p>We refer the reader to <a href="https://future.futureverse.org/reference/plan.html" class="external-link uri">https://future.futureverse.org/reference/plan.html</a> and the future backend to read up on on how to set up a plan that might work best on their setup.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicklas Stanislaus, Markus Nielsen, Christian Mains.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
