<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="An overview of the method used to simulate genotypes and case-control phenotypes.">
<title>Simulation and Liability Modelling • RyouSick</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Simulation and Liability Modelling">
<meta property="og:description" content="An overview of the method used to simulate genotypes and case-control phenotypes.">
<meta property="og:image" content="https://best-nicklas.github.io/Dataprojekt/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">RyouSick</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/RyouSick.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../articles/index.html">Articles</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="Simulation_Basis_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Simulation and Liability Modelling</h1>
            
      
      
      <div class="d-none name"><code>Simulation_Basis.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="simulating-genotypes">Simulating Genotypes<a class="anchor" aria-label="anchor" href="#simulating-genotypes"></a>
</h2>
<p>Genetic data such as DNA can be represented in various ways. The most common way to do it is to list the sequence of nucleotides (A - adenine, G - guanine, C - cytosine, T - thymine) that appear at each position of the DNA. These always appear in pairs and an example of such a sequence could be AAGGTTCGTTCT. In the RyouSick package instead of working with character sequences like these, we go straight to working with a more processed version of the sequence composed of single nucleotide polymorphisms (abbreviated SNPs and pronounced snips). SNPs give an indication of the genetic variation at a single base position in the DNA. Since the nucleotides appear in pairs any deviation from the reference genome at a given position can be represented using either a 0, 1 or 2. Let’s say that a pair in the human reference genome at a specific position is TT. If our target individual has TT at the same position in his genome we write that the SNP is 0, if he has TC (one deviation) we write 1 and if he has CC (2 deviations) we write 2. As such we would be able to represent the genomic sequence as something like 001201, only taking up half the storage space. We refer to such a sequence as the individual’s genotypes.</p>
<p>Each SNP has a certain probability of deviating from the reference. This is known as the minor allele frequency or MAF. To simulate individuals in our package we sample the MAF for each SNP from a uniform distribution so that <span class="math inline">\(MAF_j \sim uni_j(0.01, 0.49)\)</span> (although options are given to input a custom MAF vector). The values <span class="math inline">\(x_j\)</span> for each SNP position in a genotype is then sampled from a binomial distribution so <span class="math inline">\(x_j \sim binom(2, MAF_j)\)</span>.</p>
</div>
<div class="section level2">
<h2 id="liability-threshold-modeling">Liability Threshold Modeling<a class="anchor" aria-label="anchor" href="#liability-threshold-modeling"></a>
</h2>
<p>When assigning a disease status, also known as case-control phenotype, to our genotypes we make use of liability threshold modeling. The disease liability, which is used to determine an individual’s likelihood of having a disease, is modeled using two components, a genetic and an environmental liability . The genetic component is distributed as <span class="math inline">\(l_g \sim N(0, h^2)\)</span> , where <span class="math inline">\(h^2\)</span> is the heritability of a given disease and the environmental factor (the noise) as <span class="math inline">\(l_e\sim N(0,1-h^2)\)</span>. As such the full liability of an individual sums to <span class="math inline">\(l_o = l_g + l_e \sim N(0,1)\)</span>. While the environmental component can be sampled, we calculate the genetic liability using <span class="math inline">\(X^T \beta\)</span>, where X is the individual’s genotype and <span class="math inline">\(\beta\)</span> is the effect size of each SNP. The <span class="math inline">\(\beta\)</span> value for each SNP is sampled from <span class="math inline">\(N(0,\frac{h^2}{c})\)</span> when we have a causal SNP, where c is the number of causal SNPs in our target sequence, and is 0 whenever a SNP is not causal. When the full liability has been calculated we assign a case control status to the individual based on the prevalence of the disease in our population. Say there is a prevalence of 5% in the population, we then determine the threshold for being case as the upper 5% quantile of the <span class="math inline">\(l_o \text{~} N(0,1)\)</span>, e.g. 1.645. Any genotype with a lf over 1.645 will then be assigned case and all others not case.</p>
</div>
<div class="section level2">
<h2 id="simulating-family-genotypes">Simulating Family Genotypes<a class="anchor" aria-label="anchor" href="#simulating-family-genotypes"></a>
</h2>
<p>To simulate the children and siblings in our functions we first simulate two parent genotypes and find their average genotype. Different children can then be generated from these average genotypes by sampling randomly from <span class="math inline">\(\{1,2\}\)</span> whenever a SNP is 1.5 or <span class="math inline">\(\{0,1\}\)</span> whenever a SNP is 0.5. At the same time we also look for cases where both parents have a 1 at a SNP in which case we sample randomly from <span class="math inline">\(\{0,1,2\}\)</span>. Case-control status is then assigned using liability threshold modeling.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Nicklas Stanislaus, Markus Nielsen, Christian Mains.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
